### PR TITLE
Coveo headless

### PR DESCRIPTION
**Pull request by VladUsenko26**
[Coveo | Headless site example](https://linear.app/uniform/issue/INT-399/coveo-or-headless-site-example)

### CODE REVIEW

File: examples/coveo-starter/components/FacetBreadcrumbs.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+export interface FacetBreadcrumbsProps {
+  facetBreadcrumbs?: {
+    facetBreadcrumbsConfiguration?: boolean;
+  };
+}
+
+const FacetBreadcrumbs: FC<FacetBreadcrumbsProps> = ({ facetBreadcrumbs }) => {
+  if (!facetBreadcrumbs?.facetBreadcrumbsConfiguration) {
+    return <></>;
```
Comment (trsergey25) [inline examples/coveo-starter/components/FacetBreadcrumbs.tsx:48]: return null?

File: examples/coveo-starter/components/PageComposition.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import componentResolver from "@/components/componentResolver";
+import { theme } from "../context/theme";
+
+export interface PageCompositionProps {
+  data: RootComponentInstance;
+}
+
+const PageComposition: FC<PageCompositionProps> = ({ data: composition }) => {
+  return (
+    <CssBaseline>
```
Comment (trsergey25) [inline examples/coveo-starter/components/PageComposition.tsx:16]: without return

File: examples/coveo-starter/components/QuerySummary.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import {FC, useEffect, useMemo, useState} from "react";
```
Comment (trsergey25) [inline examples/coveo-starter/components/QuerySummary.tsx:1]: space and check prettier

File: examples/coveo-starter/components/componentResolver.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  "coveo-resultList": ResultList,
+  "coveo-resultsPerPage": ResultsPerPage,
+  "coveo-initial": InitialContainer,
+  "coveo-container": CoveoContainer,
+  "coveo-results-container": ResultsContainer,
+  "coveo-summary-container": SummaryContainer,
+  "coveo-facet-container": FacetsContainer,
+};
+
+export function componentResolver(
```
Comment (trsergey25) [inline examples/coveo-starter/components/componentResolver.ts:38]: I don't know if this is necessary here, but we have been using a new method of registration for half a year now 

https://docs.uniform.app/docs/learn/tutorials/csk/create-component#create-front-end-component
https://docs.uniform.app/docs/learn/tutorials/nextjs-app-router#7-implement-front-end-components

**Review by trsergey25 — APPROVED** [review]

**Review by trsergey25 — APPROVED** [review]

File: examples/coveo-starter/components/componentResolver.ts (hunk 4 above)
Comment (VladUsenko26) [reply examples/coveo-starter/components/componentResolver.ts]: thank you

**Pull request merged**